core/message_pool.go

```diff
@@ -38,21 +63,22 @@ type BlockTimer interface {
 type MessagePool struct {
 	lk sync.RWMutex
```

Comment:
fwiw i dont see this lock used as a rwlock, i see it used as a regular exclusive lock

Reply:
If you are going to change this can you give it a little [hat](https://dmitri.shuralyov.com/idiomatic-go/entries/2) -- something like `poolMu`?